src/components/core/toast/toast.js

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+import { Transition } from '@headlessui/react';
+
+import { markToastAsExiting } from './store';
+import CloseIcon from '@/components/svg/close.svg';
+
+const toastTypeClasses = {
+  info: 'bg-blue-500',
+  success: 'bg-green-500',
+  warning: 'bg-yellow-500',
+  error: 'bg-red-500',
+};
+
+/**
+ * Toast Component
+ * Manages its own lifecycle, timer, and fade out animation
+ */
+const Toast = ({ toast, onRemove }) => {
+  const timerRef = useRef(null);
+  const remainingDurationRef = useRef(0);
+  const startTimeRef = useRef(0);
+  const { id, type, message, duration, showCloseButton, isExiting } = toast;
+
+  const handleDismiss = useCallback(() => {
+    // Mark as exiting to start fade out animation
+    markToastAsExiting(id);
```

Comment:
Could we also pass `markToastAsExiting` as a prop instead of importing it from the store module?